### PR TITLE
Introduce sla_miss metric

### PR DESCRIPTION
**Pull request by Jorricks**
We have a metric for both sla callback failures and sla email failures.
However, we are not tracking when an SLA is not met.
This is crucial to know how many SLAs we had in total in statsd/prometheus.

Update: Forgot to install pre-commit :) Will fix it in a bit.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/dag_processing/test_processor.py
Comment (o-nikolas) [inline tests/dag_processing/test_processor.py:233]: Super small nit: There are other tests that already mock the stats object (for testing other SLA related metrics) that you could add this line to, instead of adding a net-new mock object to this test. But it's not a big deal, still approving. Thanks for the commit :)

**Review by o-nikolas — APPROVED** [review]

File: tests/dag_processing/test_processor.py
Comment (Jorricks) [reply tests/dag_processing/test_processor.py:233]: Yes I thought about this but I choose this test on purpose as this is one of the only ones(if not the only one) that creates the SLA by the scheduler code (instead of creating it inside the testing code). \
Therefore I feel like this was the best way to do it.

**Comment by o-nikolas** [comment]
Just looking through open PRs I've reviewed. Is there any development waiting for this one @Jorricks? Or is it just awaiting a review from a committer?

**Comment by potiuk** [comment]
needs at least rebase to fix conflict

**Comment by eladkal** [comment]
@Jorricks can you rebase and resolve conflicts?

**Comment by Jorricks** [comment]
Hey,
Sorry I dropped the ball on this one. Had some other projects going on.
Rebased and fixed conflict :)

**Comment by potiuk** [comment]
Rebased again @Jorricks - we had some "Werkzeug" drama this morning.  But I am afraid some tests that failed will fail again and will need to be fixed :(

**Comment by Jorricks** [comment]
@potiuk could you please help me out here?
I don't think the error I saw was actually related to my PR but I am not sure. Could you check to be sure?
I rebased my branch in the meantime to see if that fixes it.

**Comment by potiuk** [comment]
Likely it was intermittent failure

**Comment by potiuk** [comment]
MsSQL is not the most "stable" part of our CI.

**Comment by Jorricks** [comment]
> MsSQL is not the most "stable" part of our CI.

So should we merge this then?

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Why not :)

**Pull request merged**
